README.md

```diff
@@ -1,16 +1,29 @@
-This crate provides fuzzy search/string matching using N-grams.
+# Ngrammatic
+[![Build status](https://github.com/compenguy/ngrammatic/actions/workflows/clippy.yml/badge.svg)](https://github.com/compenguy/ngrammatic/actions)
+[![Crates.io](https://img.shields.io/crates/v/ngrammatic.svg)](https://crates.io/crates/ngrammatic)
+[![Documentation](https://docs.rs/ngrammatic/badge.svg)](https://docs.rs/ngrammatic/)
 
-This implementation is character-based, rather than word based,
-matching solely based on string similarity.
+This crate provides fuzzy search/string matching using N-grams.
 
-Licensed under the MIT license.
+This implementation is character-based, rather than word based, matching
+solely based on string similarity. It is modelled somewhat after the
+[python ngram module](https://pythonhosted.org/ngram/ngram.html) with some inspiration from
+[chappers' blog post on fuzzy matching with ngrams](http://chappers.github.io/web%20micro%20log/2015/04/29/comparison-of-ngram-fuzzy-matching-approaches/).
 
+The crate is implemented in three parts: the `Corpus`, which is an
+index connecting strings (words, symbols, whatever) to their `Ngrams`,
+and `SearchResult`s, which contains a fuzzy match result, with the
+word and a similarity measure in the range of 0.0 to 1.0.
 
-### Documentation
+The general usage pattern is to construct a `Corpus`, `.add()` your
+list of valid symbols to it, and then perform `.search()`es of valid,
+unknown, misspelled, etc symbols on the `Corpus`. The results come
+back as a vector of up to 10 results, sorted from highest similarity
```

Comment:
The `up to 10 results` bit is no longer accurate as of the earlier commit in this PR

Reply:
Updated - I am now focusing on reducing memory usage.